File: recovered/src/com/lumiyaviewer/lumiya/utils/InternPool.java
package com.lumiyaviewer.lumiya.utils;

import java.lang.ref.WeakReference;
import java.util.WeakHashMap;

/* loaded from: classes.dex */
public class InternPool<T> {
    private final WeakHashMap<T, WeakReference<T>> pool = new WeakHashMap<>();

    public final synchronized T intern(T t) {
        T t2;
        WeakReference<T> weakReference = this.pool.get(t);
        if (weakReference != null && (t2 = weakReference.get()) != null) {
            return t2;
        }
        this.pool.put(t, new WeakReference<>(t));
        return t;
    }
}

File: app/src/main/java/com/lumiyaviewer/lumiya/utils/InternPool.kt
package com.lumiyaviewer.lumiya.utils

import java.lang.ref.WeakReference
import java.util.WeakHashMap

open class InternPool<T> {
    private val pool = WeakHashMap<T, WeakReference<T>>()
    @Synchronized fun intern(value: T): T {
        pool[value]?.get()?.let { return it }
        pool[value] = WeakReference(value)
        return value
    }
}
